### PR TITLE
Cleanup peer handshake data on close stream.

### LINKED ISSUE #1177: delete notifications protocol handshake data on stream close
## Expected Behavior

- if a stream closes w/ a notifications protocol ID we should delete the handshake data for that peer, since we need to redo the handshake if the stream re-opens
- can use `ConnManager.ClosedStream` for this

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [ ] I am planning to submit a pull request to fix this issue myself

### PR DESCRIPTION
**Pull request by arijitAD**
## Changes

- Cleanup peer handshake data on close stream.

## Tests

```
Manually tested:
Started two gossamer nodes and killed one of them after the notification protocol handshake.
```

## Checklist

- [x] I have read [CODE_OF_CONDUCT](https://github.com/ChainSafe/gossamer/blob/development/.github/CODE_OF_CONDUCT.md) and [CONTRIBUTING](https://github.com/ChainSafe/gossamer/blob/development/.github/CONTRIBUTING.md) 
- [x] I have provided as much information as possible and necessary
- [x] I have reviewed my own pull request before requesting a review
- [x] All integration tests and required coverage checks are passing

## Issues

- closes #1177

### CODE REVIEW

File: dot/network/connmgr.go [hunk 1]
```diff
@@ -146,5 +160,11 @@ func (cm *ConnManager) ClosedStream(n network.Network, s network.Stream) {
 		"protocol", s.Protocol(),
 	)
 
-	// TODO: on stream close, clear notifications protocol data for this peer
+	cm.Lock()
+	defer cm.Unlock()
```
Comment (noot) [inline dot/network/connmgr.go:164]: this is nice!

**Review by noot — COMMENTED** [review]
looks good! could you add a unit test for this? you would need to start 2 nodes then open a stream between them on some sub-protocol (probably block announce) then close the stream and check if the handshake data is deleted. you can use the function `getStream` in `host.go` to get the open stream

**Comment by arijitAD** [comment]
> looks good! could you add a unit test for this? you would need to start 2 nodes then open a stream between them on some sub-protocol (probably block announce) then close the stream and check if the handshake data is deleted. you can use the function `getStream` in `host.go` to get the open stream

Added unit tests.

File: dot/network/host_test.go [hunk 2]
```diff
@@ -439,3 +440,83 @@ func TestExistingStream(t *testing.T) {
...
+
+	stream := nodeA.host.getStream(nodeB.host.id(), "")
+	require.Nil(t, stream, "node A should not have an outbound stream")
+
+	// node A opens the stream to send the first message
+	err = nodeA.host.send(addrInfosB[0].ID, "", TestMessage)
+	require.NoError(t, err)
+
+	info := nodeA.notificationsProtocols[BlockAnnounceMsgType]
+	handler := nodeA.createNotificationsMessageHandler(info, nodeA.validateBlockAnnounceHandshake, nodeA.handleBlockAnnounceMessage)
```
Comment (noot) [inline dot/network/host_test.go:494]: the service already creates and registers this in `service.Start` so I think you can remove this

File: dot/network/host_test.go [hunk 3]
```diff
@@ -439,3 +440,83 @@ func TestExistingStream(t *testing.T) {
...
+	}
+	msg :=  &BlockAnnounceHandshake{
+		Roles:           4,
+		BestBlockNumber: 77,
+		BestBlockHash:   common.Hash{1},
+		GenesisHash:     common.Hash{2},
+	}
+
+	err = handler(nodeB.host.id(), msg)
+	require.NoError(t, err)
```
Comment (noot) [inline dot/network/host_test.go:509]: same here, the message should automatically be handled

**Review by noot — APPROVED** [review]
looks good!

**Review by RyRy79261 — APPROVED** [review]

**Pull request merged**